lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -2390,6 +2390,43 @@ bool TypeChecker::isExplicitlyConvertibleTo(Type type1, Type type2,
                                 ConstraintKind::ExplicitConversion, dc);
 }
 
+bool TypeChecker::isDeclConvertibleTo(const ValueDecl *decl, Type fromType,
+                                      Type toType) {
+  auto *outerDC = decl->getDeclContext();
+  auto *innerDC = decl->getInnermostDeclContext();
+
+  ConstraintSystem cs(*this, outerDC, ConstraintSystemOptions());
+
+  auto openType = [innerDC, outerDC, &cs](Type type) -> Type {
+    if (type->is<GenericFunctionType>()) {
+      llvm::DenseMap<CanType, TypeVariableType *> replacements;
+      return cs.openFunctionType(type->castTo<GenericFunctionType>(),
+                                 /*numArgLabelsToRemove=*/0,
+                                 /*locator=*/nullptr, replacements, innerDC,
+                                 outerDC, /*skipProtocolSelfConstraint=*/false);
+    } else if (type->hasTypeParameter()) {
+      return cs.openType(type, /*locator=*/nullptr);
```

Comment:
I guess `openGeneric` would only really apply to `fromType` in this manner. Still, it would work better for properties/subscripts, if they are using this code path.